blog/2023-11-01-feature-flags-sdks-architectures.md

```diff
@@ -0,0 +1,159 @@
+---
+slug: 'feature-flags-sdks-architectures'
+title: 'Feature Flags services Server-side SDKs different architectures approaches'
+date: 2023-11-01
+authors: [liran2000]
+description: "Feature Flags services Server-side SDKs different architectures approaches"
+tags: [client, sdk]
+draft: false
+---
+
+Server-side feature flag software development kits (SDKs) are common for letting applications like microservices interact with Feature flag services.
+Feature Flags SDKs have several used functionalities, mainly feature flag evaluation with or without context.
+Each Feature Flags service can publish SDKs in multiple programming languages.
+The Feature Flags services commonly expose APIs via endpoints like REST HTTP and/or gRPC.
+SDKs are an important layer, as network traffic loads affect both the application and the Feature Flags services.
+When the Feature Flags service is a Cloud SaaS service, some services support a relay proxy or a sidecar application, which can be deployed on an organization network.
+The Relay Proxy lets multiple clients connect to a local proxy, reducing the number of outbound connections to the Cloud Service.
+Considering that a large amount of microservices using SDK instances can be deployed, this can be significant.
+
+
+<!--truncate-->
+
+## Feature Flags services Server-side SDKs different architectures approaches
+
+During usage and development involvement at several feature flag services providers, I have encountered some different
+architectures approaches used by some vendors. Each vendor server-side SDK can commonly belong to one of the described
+approaches.
+Here some of the main server-side SDKs functionalities is described, however, there are additional functionalities like
+statistics and analytics reporting, metrics, events and more.
+
+### Notes / Glossary
+
+* Described architectures are at high level.
+* Features Service described at the diagrams can be a Cloud service, a local network Proxy or some data source
+  (some feature flags solutions SDKs can communicate directly from a source like Github).
+* There are additional functionalities like statistics reporting, metrics, events and more.
+
```

Comment:
Can we change Lines 33 - 35 to read:
* The described architectures below are high level.
* Feature flag services described in the diagrams may be a Cloud service, a local network Proxy or some data source (some feature flags solutions SDKs can communicate directly from a source like Github).

For line 36 - are you saying that these are additional functionalities that you will not cover here? If so, we should make that explicitly clear by saying something to that effect (like below): 
* Additional functionalities like statistics reporting, metrics, events and more should also be taken into consideration, but I have not explored these in this particular blog post.